### PR TITLE
Support customizing image family name.

### PR DESCRIPTION
**Pull request by canliu-aha**
b/315387164

TESTED: manually run with --image-family-name flag.
Screenshot 2024-01-09 at 5 56 26 PM

### CODE REVIEW

File: gke-disk-image-builder/cli/main.go [hunk 1]
```diff
@@ -37,10 +38,15 @@ func (s *stringSlice) Set(value string) error {
...
 
+var (
+	gcpResourceNameRegex = regexp.MustCompile("^[a-z]([-a-z0-9]*[a-z0-9])?$")
+)
+
 func main() {
 	var containerImages stringSlice
 	projectName := flag.String("project-name", "", "name of a gcp project where the script will be run")
 	imageName := flag.String("image-name", "", "name of the image that will be generated")
+	imageFamilyName := flag.String("image-family-name", "secondary-disk-image", "name of the image family associated with the created disk image")
```
Comment (danielvegamyhre) [inline gke-disk-image-builder/cli/main.go:49]: Is "secondary-disk-image" a valid image family name? I haven't heard of it, if it's not let's not set it as a default.

File: gke-disk-image-builder/cli/main.go [hunk 2]
```diff
@@ -37,10 +38,15 @@ func (s *stringSlice) Set(value string) error {
 	return nil
 }
 
+var (
+	gcpResourceNameRegex = regexp.MustCompile("^[a-z]([-a-z0-9]*[a-z0-9])?$")
```
Comment (danielvegamyhre) [inline gke-disk-image-builder/cli/main.go:42]: Where did you find the rules for GCP resource names?

Comment (canliu-aha) [reply gke-disk-image-builder/cli/main.go:42]: https://cloud.google.com/compute/docs/naming-resources

The regex is here: https://screenshot.googleplex.com/5H4GFNpxnDabQJX

File: gke-disk-image-builder/cli/main.go (hunk 1 above)
Comment (canliu-aha) [reply gke-disk-image-builder/cli/main.go:49]: It is the image family name currently hardcoded in the disk image builder code, so I'm using it as the default. Shall I change to something else?

Comment (danielvegamyhre) [reply gke-disk-image-builder/cli/main.go:49]: Oh weird, I never noticed that, ok sounds good.

**Review by danielvegamyhre — APPROVED** [review]

**Comment by danielvegamyhre** [comment]
@canliu-aha Before I merge this, can you confirm you have manually tested these changes?

**Comment by canliu-aha** [comment]
Yes, I confirmed I've manually tested the behavior.

**Pull request merged**